x-pack/plugins/observability_solution/slo/server/services/slo_repository.ts

```diff
@@ -73,7 +73,7 @@ export class KibanaSavedObjectsSLORepository implements SLORepository {
     return slo;
   }
 
-  async deleteById(id: string): Promise<void> {
+  async deleteById(id: string, ignore404 = false): Promise<void> {
```

Comment:
I know what you mean by ignore404, but that's not accurate: we are not handling a 404 http status code. Can you use something similar to the save method? e.g. 
`options?: { ignoreNotFound: boolean }` or `options?: { ignoreInexistant: boolean }` or even inverse the default behaviour, but that would be more work i guess: `options?: { throwOnInexistant: boolean }`